src/App.tsx

```diff
@@ -20,12 +22,15 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={(event) => {
+                  setValue(event.target.value);
+                }}
               />
             </div>
           </div>
+          <MoviesList movies={moviesFromServer} value={query} />
```

Comment:
why not just pass a filtered movies array?
```suggestion
const finalQuery = value.trim().toLocaleLowerCase();
const filteredMovies = moviesFromServer.filter(...
  (title + description).includes(finalQuery)
)
...
          <MoviesList movies={filteredMovies} value={query} />
```

then i guess `MovieCard.tsx` would need no changes at all

Reply:
Hello @meforoma, ok I will apply your suggestion but I don't understand what is the difference? these are two different ways, does the way you are suggesting have some advantages? :) thank you

Reply:
well, yes, the main advantage is
- if you `.map` 5 children components,
- but decide to render them or not inside the child component itself
- it's much less optimal compared to `.map` a list of alreade pre-filtered children components

Reply:
Ohh I see, got it, thanks @meforoma for your explanation :)